wgsl/index.bs

```diff
@@ -5274,6 +5274,99 @@ A phony-assignment is useful for:
   </xmp>
 </div>
 
+### Complex assignment ### {#complex-assignment-sec}
+
+A <dfn noexport>complex assignment</dfn> is a statement that combines an arithmetic operation and an assignment.
+
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>complex_assignment_statement</dfn> :
+
+    | [=syntax/lhs_expression=] [=syntax/plus_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/minus_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/times_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/division_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/modulo_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/and_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/or_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/xor_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/shift_right_equal=] [=syntax/short_circuit_or_expression=]
+
+    | [=syntax/lhs_expression=] [=syntax/shift_left_equal=] [=syntax/short_circuit_or_expression=]
+</div>
+
+The type requirements, semantics, and behavior of each statement is defined as if
+it expands as in the following table, except that:
+* The reference expression |e1| is evaluated only once.
+* The identifier `old` is chosen to be different from any other identifier in the program.
+
+<table class='data'>
+  <thead>
+    <tr><th>Statement<th>Expansion
+  </thead>
+<tr algorithm="add-assign">
+    <td class="nowrap">|e1| += |e2|
+    <td>let old = |e1|; |e1| = old + (|e2|)
```

Comment:
You've already said that the reference expression `|e1|` is only evaluated once (above), so why do we need to introduce this temporary `old` value as well?
I think `|e1| = |e1| + (|e2|)` is sufficient.

Reply:
I was worried about whether someone would get confused about  "Evaluating |e1|" meaning the reference vs. the reference-and-the-load-rule.

Maybe it's all made clear with the example, which I added after drafting the text.  I'll revert this bit, and so it will get much closer to Robin's initial PR.